src/peft/tuners/lora/layer.py

```diff
@@ -139,6 +146,18 @@ def reset_lora_parameters(self, adapter_name):
             nn.init.zeros_(self.lora_embedding_A[adapter_name])
             nn.init.normal_(self.lora_embedding_B[adapter_name])
 
+    def scale_layer(self, scale_factor: float) -> None:
+        if scale_factor != 0 and scale_factor != 1:
```

Comment:
Should we really just do nothing if `scale_factor == 0`? Maybe users really want to set it to 0? Or if they shouldn't do it (we can't unscale later), raise an error?

Reply:
addressed in the latest commit.